src/app/modules/customer-management/store/customer-management.reducers.ts

```diff
@@ -58,10 +58,75 @@ export function customerManagementReducer(
     case CustomerManagementActionTypes.CREATE_CUSTOMER_FAIL: {
       return {
         ...state,
-        data: [],
+        data: state.data,
         isLoading: false,
         message: 'An error occurred, try again later.',
```

Comment:
you need to reset to '' the activityIdToEdit variable in all fail cases.

Reply:
done